README.markdown

```diff
@@ -55,17 +52,37 @@ Rack::Timeout.timeout = 5
 ```ruby
 # config.ru
 require "rack-timeout"
-use Rack::Timeout          # Call as early as possible so rack-timeout runs before all other middleware.
-Rack::Timeout.timeout = 5  # Recommended. If omitted, defaults to 15 seconds.
+
+# Call as early as possible so rack-timeout runs before all other middleware.
+# Setting service_timeout is recommended. If omitted, defaults to 15 seconds.
+use Rack::Timeout, service_timeout:5
 ```
 
 
 The Rabbit Hole
 ---------------
 
+Rack::Timeout has four settings, with the following defaults:
+
+```
+service_timeout:   15
+wait_timeout:      30
+wait_overtime:     60
+service_past_wait: false
+```
+
+As shown earlier, these settings can be overriden during middleware initialization:
+
+```ruby
+use Rack::Timeout, service_timeout:5, wait_timeout:false
+```
+
+For legacy reasons, it's also possible to set these at the class level (e.g. `Rack::Timeout.service_timeout = 3`). This is disrecommended tho. Also beware this style takes precedence over all instances' initialization settings.
+
+
 ### Service Timeout
 
-`Rack::Timeout.timeout` (or `Rack::Timeout.service_timeout`) is our principal setting.
+`service_timeout` is our most important setting.
```

Comment:
I kinda agree, but because the settings are a bit of a shit show, I also find it important to have something like "if you can only care about one thing, this is the setting you have to pay attention to"